src/nxt_http_return.c

```diff
@@ -87,27 +88,130 @@ nxt_http_return(nxt_task_t *task, nxt_http_request_t *r,
         return NULL;
     }
 
+    ctx = nxt_mp_zget(r->mem_pool, sizeof(*ctx));
+    if (nxt_slow_path(ctx == NULL)) {
+        goto fail;
+    }
+
+    ctx->action = action;
     r->status = conf->status;
     r->resp.content_length_n = 0;
 
-    if (conf->location.length > 0) {
+    if (conf->loc_is_const) {
+        enc = nxt_str_dup(r->mem_pool, &ctx->encoded, &conf->encoded);
```

Comment:
I suggest that.
```
ctx->encoded = conf->encoded;
```

Reply:
Yup.  I didn't know if I could do that.  If you see any other copies that can be replaced by a simple assignment, please tell me.  When do I need to `nxt_str_dup()`?

Reply:
For me, I usually check it by their life cycle and if it will be changed. For example, `conf->encoded` has a longer life cycle than `ctx->encoded`, and `ctx->encoded` doesn't change after that.

Reply:
Thanks.  I'll replace the other one in line 144 too.